nomad-base/src/settings/mod.rs

```diff
@@ -164,20 +85,31 @@ pub struct IndexSettings {
 }
 
 impl IndexSettings {
-    /// Get the `from` setting
-    pub fn from(&self) -> u32 {
-        self.from
-            .as_ref()
-            .and_then(|s| s.parse::<u32>().ok())
-            .unwrap_or_default()
-    }
-
-    /// Get the `chunk_size` setting
-    pub fn chunk_size(&self) -> u32 {
-        self.chunk
-            .as_ref()
-            .and_then(|s| s.parse::<u32>().ok())
-            .unwrap_or(1999)
+    /// Get agent-specific index settings unique to that agent
+    pub fn from_agent_name(agent_name: &str) -> Self {
```

Comment:
future: this should be type-associated information on the agent type. open an issue here

Reply:
https://github.com/nomad-xyz/rust/issues/43